src/auth_webhook.py

```diff
@@ -0,0 +1,184 @@
+import json
+import logging
+import os
+import random
+import re
+import string
+import tempfile
+from base64 import b64decode, b64encode
+from subprocess import CalledProcessError, check_call, check_output
+
+import charms.contextual_status as status
+import yaml
+from jinja2 import Environment, FileSystemLoader
+from kubectl import kubectl
+from ops import MaintenanceStatus
+
+auth_secret_ns = "kube-system"
+auth_secret_type = "juju.is/token-auth"
+auth_webhook_root = "/root/cdk/auth-webhook"
+auth_webhook_conf = os.path.join(auth_webhook_root, "auth-webhook-conf.yaml")
+auth_webhook_exe = os.path.join(auth_webhook_root, "auth-webhook.py")
+# wokeignore:rule=master
+auth_webhook_svc_name = "cdk.master.auth-webhook"
+auth_webhook_svc = "/etc/systemd/system/{}.service".format(auth_webhook_svc_name)
+log_dir = "/var/log/kubernetes"
+
+log = logging.getLogger(__name__)
+
+
+def configure(
+    charm_dir, aws_iam_endpoint=None, custom_authn_endpoint=None, keystone_endpoint=None
+):
+    """Render auth webhook templates and start the related service."""
+    status.add(MaintenanceStatus("Configuring auth webhook"))
+
+    # Set the number of gunicorn workers based on our core count. (2*cores)+1 is
+    # recommended: https://docs.gunicorn.org/en/stable/design.html#how-many-workers
+    try:
+        cores = int(check_output(["nproc"]).decode("utf-8").strip())
+    except CalledProcessError as e:
+        log.exception(e)
+        log.error("Failed to determine core count for auth-webhook")
+        # Our default architecture is 2-cores for k8s-cp units
+        cores = 2
+    else:
+        # Put an upper bound on cores; more than 12ish workers is overkill
+        cores = 6 if cores > 6 else cores
+
+    # For 'api_ver', match the api version of the authentication.k8s.io TokenReview
+    # that k8s-apiserver will be sending:
+    #   https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18
+    context = {
+        "api_ver": "v1beta1",
```

Comment:
```suggestion
    #   https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#tokenreview-v1-authentication-k8s-io
    context = {
        "api_ver": "v1",
```
`apiVersion: v1` has been available for a while, but i'm ok to ignore for now if you'd rather keep the focus of these PRs to conversion rather than conversion + new changes.

Reply:
Heck yeah, let's clean it up. Thanks for pointing it out.